crates/sui-surfer/src/surfer_state.rs

```diff
@@ -0,0 +1,393 @@
+            effects.status().is_ok(),
+            package,
+            module,
+            function,
+        );
+        self.process_tx_effects(&effects).await;
+    }
+
+    async fn process_tx_effects(&mut self, effects: &SuiTransactionBlockEffects) {
+        for (owned_ref, write_kind) in effects.all_changed_objects() {
+            let obj_ref = owned_ref.reference.to_object_ref();
+            let object = self
+                .cluster
+                .get_object_from_fullnode_store(&obj_ref.0)
+                .await
+                .unwrap();
+            if object.is_package() {
+                self.discover_entry_functions(object).await;
+                continue;
+            }
+            let struct_tag = object.struct_tag().unwrap();
+            match owned_ref.owner {
+                Owner::Immutable => {
+                    self.immutable_objects
+                        .write()
+                        .await
+                        .entry(struct_tag)
+                        .or_default()
+                        .push(obj_ref);
+                }
+                Owner::AddressOwner(address) => {
+                    if address == self.address {
+                        self.owned_objects
+                            .entry(struct_tag)
+                            .or_default()
+                            .insert(obj_ref);
+                    }
+                }
+                Owner::ObjectOwner(_) => (),
+                Owner::Shared {
+                    initial_shared_version,
+                } => {
+                    if write_kind != WriteKind::Mutate {
+                        self.shared_objects
+                            .write()
+                            .await
+                            .entry(struct_tag)
+                            .or_default()
+                            .push((obj_ref.0, initial_shared_version));
+                    }
+                    // We do not need to insert it if it's a Mutate, because it means
+                    // we should already have it in the inventory.
+                }
+            }
+            if obj_ref.0 == self.gas_object.0 {
+                self.gas_object = obj_ref;
+            }
+        }
+    }
+
+    async fn discover_entry_functions(&self, package: Object) {
+        let package_id = package.id();
+        let move_package = package.data.try_into_package().unwrap();
+        let move_binary_format_version =
+            ProtocolConfig::get_for_max_version().move_binary_format_version();
+        let entry_functions: Vec<_> = move_package
+            .normalize(move_binary_format_version)
+            .unwrap()
+            .into_iter()
+            .flat_map(|(module_name, module)| {
+                module
+                    .functions
+                    .into_iter()
+                    .filter_map(|(func_name, func)| {
+                        if !matches!(func.visibility, Visibility::Public) {
```

Comment:
There are private and friend `entry` functions you can call too in a transaction. 
https://github.com/MystenLabs/sui/blob/ddfe4ff71a7e891c954dff57de84de477ac36979/crates/sui-adapter/src/programmable_transactions/execution.rs#L984-L986

Reply:
Fixed